python/mxnet/module/executor_group.py

```diff
@@ -421,17 +421,31 @@ def get_params(self, arg_params, aux_params):
             Target parameter arrays.
         aux_params : list of NDArray
             Target aux arrays.
+        copy_to_cpu : boolean
+            Whether or not to copy parameters to CPU. (default to 'true')
 
         Notes
         -----
         - This function will inplace update the NDArrays in arg_params and aux_params.
         """
-        for name, block in zip(self.param_names, self.param_arrays):
-            weight = sum(w.copyto(ctx.cpu()) for w in block) / len(block)
-            weight.astype(arg_params[name].dtype).copyto(arg_params[name])
-        for name, block in zip(self.aux_names, self.aux_arrays):
-            weight = sum(w.copyto(ctx.cpu()) for w in block) / len(block)
-            weight.astype(aux_params[name].dtype).copyto(aux_params[name])
+        if copy_to_cpu:
+            for name, block in zip(self.param_names, self.param_arrays):
+                weight = sum(w.copyto(ctx.cpu()) for w in block) / len(block)
+                weight.astype(arg_params[name].dtype).copyto(arg_params[name])
+            for name, block in zip(self.aux_names, self.aux_arrays):
+                weight = sum(w.copyto(ctx.cpu()) for w in block) / len(block)
+                weight.astype(aux_params[name].dtype).copyto(aux_params[name])
+        else:
+            for name, block in zip(self.param_names, self.param_arrays):
+                context = block[0].context
```

Comment:
It seems this `else` block is identical to the `if` block except for the context. Can you add the if check to the context and change context between `block[0].context` and `ctx.cpu()`

Reply:
Fixed